core/shared/src/main/scala/fs2/hashing/Hashing.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2
+package hashing
+
+import cats.effect.Sync
+
+/** Capability trait that provides hashing.
+  *
+  * The [[create]] method returns an action that instantiates a fresh `Hash` object.
+  * `Hash` is a mutable object that supports incremental computation of hashes. A `Hash`
+  * instance should be created for each hash you want to compute.
+  */
+trait Hashing[F[_]] {
+  def create(algorithm: String): F[Hash[F]]
```

Comment:
Ideally this should return a `Resource` so that we can do cleanup or back it by a pool.

```suggestion
  def create(algorithm: String): Resource[F, Hash[F]]
```

Reply:
Hm, [do we really need object pooling](https://stackoverflow.com/questions/13913075/to-pool-or-not-to-pool-java-crypto-service-providers)?

Reply:
Maybe, maybe not. We still need a `Resource` anyway for Scala Native, and it opens the door to pooling if someone decides that's what they want to do.

Reply:
Oh gotcha, hadn’t gotten as far as native integration yet. Makes sense that we’ll need to release something.